### PR TITLE
fix: (Core) issues on Popover related components

### LINKED ISSUE #3694: Popover Related Components - Width
I see we have different look for control element on select/combobox/multi input/datetime
comparing styles and NGX
- on ngx all of these elements got display: block and have width equals to container
- on styles all of these element have width depending on child
The example is in select/combobox/multiinput components, that should have body limited to `600px` and `control limited only to content width`
Styles:
image

NGX:
image

### PR DESCRIPTION
**Pull request by JKMarkowski**
#### Please provide a link to the associated issue.
Closes #3694
#### Please provide a brief summary of this pull request.
There are 
- changes mostly in select/combobox components regarding width of control and body. + Overflow for components with truncation. 
- fix for IE11 for multiinput - for some reason focusing input was opening multi input. which made using multiinput impossible
- datetime picker width change
- fix for bug with height in some components

Before:
Combobox:
image

DateTimePicker:
image

Select:
image

After:
Combobox:
image

DateTimePicker:
image

Select:
image

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/select/examples/select-nested-options/select-nested-options.component.html [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <fd-select [(value)]="selectedValue"
+           [extendedBodyTemplate]="true"
```
Comment (JKMarkowski) [inline apps/docs/src/app/core/component-docs/select/examples/select-nested-options/select-nested-options.component.html:2]: `extendedBodyTemplate` is added to let Option Components know if the text should be wrapped in `fd-list-title` directive. It's mostly to keep overflow in correct way

File: libs/core/src/lib/select/select.component.scss [hunk 2]
```diff
@@ -1,22 +1,17 @@
...
-    & > fd-popover {
-        width: 100%;
-
-        &.fd-popover__body--dropdown {
-            width: auto;
-            display: block;
-        }
-    }
+.fd-popover__body > :first-child {
+	border-radius: 0;
```
Comment (stefanoScalzo) [inline libs/core/src/lib/select/select.component.scss:4]: will also need to fix the border radius of the popover

File: libs/core/src/lib/combobox/combobox.component.ts [hunk 3]
```diff
@@ -171,6 +172,15 @@ export class ComboboxComponent implements ComboboxInterface, ControlValueAccesso
     @Input()
     autoComplete = true;
 
+    /**
+     * Preset options for the Combobox body width.
+     * * `at-least` will apply a minimum width to the body equivalent to the width of the InputGroup. - Default
+     * * `equal` will apply a width to the body equivalent to the width of the InputGroup.
+     * * '' for no effect
```
Comment (stefanoScalzo) [inline libs/core/src/lib/combobox/combobox.component.ts:179]: Describe what is meant by no effect please

**Review by stefanoScalzo — CHANGES_REQUESTED** [review]

**Comment by JKMarkowski** [comment]
@stefanoScalzo Your comments has been addresed, can you check it again?

File: libs/core/src/lib/select/option/option.component.ts [hunk 4]
```diff
@@ -70,6 +70,9 @@ export class OptionComponent implements OnInit, OnDestroy {
     /** @hidden */
     selectionEvent = new EventEmitter<KeyboardEvent>()
 
+    /** @hidden Whether option contains more than basic text. */
+    extendedTemplate = false;
```
Comment (InnaAtanasova) [inline libs/core/src/lib/select/option/option.component.ts:74]: if it's hidden it should be prefixed with `_` as we discussed

**Review by InnaAtanasova — APPROVED** [review]

**Comment by InnaAtanasova** [comment]
Are these related to the PR? This is Select component
Screen Shot 2020-11-05 at 11 36 09 AM
Screen Shot 2020-11-05 at 11 36 03 AM

File: libs/core/src/lib/select/option/option.component.ts (hunk 4 above)
Comment (InnaAtanasova) [reply libs/core/src/lib/select/option/option.component.ts]: You can wait till we do the refactoring as this is not a new component.

**Comment by JKMarkowski** [comment]
> Are these related to the PR? This is Select component
> Screen Shot 2020-11-05 at 11 36 09 AM

This one was there for a long time. I guess container for message shoule be shorter, but it's definetely scope of this PR

File: libs/core/src/lib/multi-input/multi-input.component.ts [hunk 5]
```diff
@@ -453,7 +455,7 @@ export class MultiInputComponent implements
 
     /** @hidden */
     moreClicked(): void {
-        this.openChangeHandle(true);
+        // this.openChangeHandle(true);
```
Comment (salarenko) [inline libs/core/src/lib/multi-input/multi-input.component.ts:458]: I think I agree this is not needed here, but should not be left like that.

File: libs/core/src/lib/multi-input/multi-input.component.ts [hunk 6]
```diff
@@ -466,13 +468,17 @@ export class MultiInputComponent implements
...
-        } else {
-            this.openChangeHandle(false);
-        }
+        this.openChangeHandle(!this.open);
     }
 
+    /** @hidden */
+    private _propagateSearchTermChange(): void {
+        this.searchTermChange.emit(this.searchTerm);
+        this.displayedValues = this.filterFn(this.dropdownValues, this.searchTerm);
```
Comment (salarenko) [inline libs/core/src/lib/multi-input/multi-input.component.ts:477]: I'm not sure if we should have `displayedValues` updated in a method which (based on the name) should be responsible for propagating some changes.

File: libs/core/src/lib/popover/popover-position/popover-position.ts [hunk 7]
```diff
@@ -25,7 +25,7 @@ export const DefaultPositions: ConnectedPosition[] = [
     { originX: 'end', originY: 'top', overlayX: 'end', overlayY: 'bottom' }
 ];
 
-export type PopoverFillMode = 'at-least' | 'equal';
+export type PopoverFillMode = 'at-least' | 'equal' | 'default';
```
Comment (salarenko) [inline libs/core/src/lib/popover/popover-position/popover-position.ts:28]: `PopoverFillMode` and the types should be described.

File: libs/core/src/lib/popover/popover.component.scss [hunk 8]
```diff
@@ -123,3 +124,13 @@ $fd-popover-background-color: var(--sapGroup_ContentBackground, #fff) !default;
         cursor: auto;
     }
 }
+
+.cdk-overlay-connected-position-bounding-box {
+	height: auto !important;
+}
+
+.fd-popover-custom-list,
```
Comment (salarenko) [inline libs/core/src/lib/popover/popover.component.scss:132]: Wasn't this "fixed" in #3739?

File: libs/core/src/lib/popover/popover.component.ts [hunk 9]
```diff
@@ -370,10 +376,12 @@ export class PopoverComponent extends BasePopoverClass
 
     /** @hidden */
     private _applyWidthOverlay(): void {
+        const maxWidthLimit = this.maxWidth ? this.maxWidth : MAX_BODY_SIZE;
```
Comment (salarenko) [inline libs/core/src/lib/popover/popover.component.ts:379]: Do we need separately set the width to overlay and on `.fd-popover__popper` level?

**Review by salarenko — CHANGES_REQUESTED** [review]
- Select in mobile mode still takes all available width
image

- I couldn't find any straight information, but I think according to the documentation Multi Input should not be growing when tokens are added. 
growing-multi-input

File: libs/core/src/lib/combobox/combobox.component.ts [hunk 10]
```diff
@@ -171,6 +172,15 @@ export class ComboboxComponent implements ComboboxInterface, ControlValueAccesso
     @Input()
     autoComplete = true;
 
+    /**
+     * Preset options for the Combobox body width.
+     * * `at-least` will apply a minimum width to the body equivalent to the width of the control. - Default
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply libs/core/src/lib/combobox/combobox.component.ts:177]: It should be added, that it's limited by list item

File: libs/core/src/lib/multi-input/multi-input.component.ts (hunk 6 above)
Comment (JKMarkowski) [reply libs/core/src/lib/multi-input/multi-input.component.ts:477]: I changed naming

File: libs/core/src/lib/popover/popover.component.scss (hunk 8 above)
Comment (JKMarkowski) [reply libs/core/src/lib/popover/popover.component.scss:132]: Good catch

File: libs/core/src/lib/combobox/combobox.component.html [hunk 11]
```diff
@@ -5,13 +5,12 @@
         additionalBodyClass="fd-popover-custom-list"
         [isOpen]="open && displayedValues && displayedValues.length"
         (isOpenChange)="isOpenChangeHandle($event)"
-        [fillControlMode]="'at-least'"
+        [fillControlMode]="fillControlMode"
         [focusTrapped]="true"
         [triggers]="triggers"
+        [maxWidth]="600"
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply libs/core/src/lib/combobox/combobox.component.html:11]: It is, but, ppover needs to know about this limit

File: libs/core/src/lib/popover/popover.component.ts (hunk 9 above)
Comment (JKMarkowski) [reply libs/core/src/lib/popover/popover.component.ts:379]: Right now styles are applied only on `.fd-popover__popper` level

**Review by salarenko — APPROVED** [review]

**Pull request merged**